src/ServicePulse.Host/vue/src/components/failedmessages/EditMessageHeader.vue

```diff
@@ -0,0 +1,70 @@
+<script setup>
+import { ref, watch, computed, onMounted } from "vue";
+
+const settings = defineProps({
+  header: Object,
+});
+
+let origHeaderValue = undefined;
+const header = ref(settings.header);
+
+const headerValue = computed(() => settings.header.value);
+watch(headerValue, async (newValue) => {
+  header.value.isChanged = newValue !== origHeaderValue ? true : false;
+});
+
+function resetHeaderChanges() {
+  header.value.value = origHeaderValue;
+  header.value.isMarkedAsRemoved = false;
+  header.value.isChanged = false;
+}
+
+function markHeaderAsRemoved() {
+  header.value.isMarkedAsRemoved = true;
+  header.value.isChanged = true;
+}
+
+onMounted(() => {
+  origHeaderValue = settings.header.value;
+});
+</script>
+
+<template>
+  <td nowrap="nowrap">
+    <span :class="{ 'header-removed': header.isMarkedAsRemoved }">{{ settings.header.key }}</span>
+    <span v-if="header.isLocked">
+      &nbsp;
+      <i class="fa fa-lock" tooltip="Protected system header"></i>
+    </span>
+    <span v-if="(header.isChanged || header.isMarkedAsRemoved) && header.isSensitive">
+      &nbsp;
+      <i class="fa fa-exclamation-triangle" uib-tooltip="This is a sensitive message header that if changed can the system behavior. Proceed with caution."></i>
+    </span>
+    <span v-if="header.isChanged">
+      &nbsp;
+      <i class="fa fa-pencil" tooltip="Edited"></i>
+    </span>
+  </td>
+  <td>
+    <input :class="{ 'header-removed': header.isMarkedAsRemoved }" class="form-control" :disabled="header.isLocked" v-model="header.value" />
```

Comment:
```suggestion
    <input :class="{ 'header-removed': header.isMarkedAsRemoved }" class="form-control" :disabled="header.isLocked" :value="header.value" />
```

Reply:
`v-model` is a two-way binding for form inputs.  By changing this to `:value` it breaks the reactivity of the input field.  I would have to add `:input` as well and call another function to get the reactivity back.  However, this has highlighted a couple of things for me:
 - I had to fetch the message data again from the ServiceControl API to reset the values on a cancel request, which might not be necessary if we only are using `:value` and `:input`. [Here is the code](https://github.com/Particular/ServicePulse/pull/1416/files#diff-045b53b9f0f278f3c9f435926328841394e39d567c341feed34754807d148295R316)
 -  Also, I watch the computed header value [here](https://github.com/Particular/ServicePulse/pull/1416/files#diff-cd3f7081646b1c0eb2421a2e3516af2c7a6eb92a42f0a73739fdb6899d5afb31R11-R13), but I think I read somewhere that getting a value from `computed()` is like a snapshot of the value and shouldn't be mutated.  However, we do state [here](https://github.com/Particular/Platform/blob/main/documentation/service-pulse.md#reactivity) that using `computed()` to watch for changes in a given value is a best practice.  So, I'm not sure if I am doing this right by binding the model to an input component like this?

Reply:
Ok, got it :+1: